src/main/java/com/timgroup/statsd/UnixStreamClientChannel.java

```diff
@@ -134,28 +135,39 @@ private void connect() throws IOException {
             // We'd have better timeout support if we used Java 16's native Unix domain socket support (JEP 380)
             delegate.setOption(UnixSocketOptions.SO_SNDTIMEO, connectionTimeout);
         }
-        if (!delegate.connect(address)) {
-            if (connectionTimeout > 0 && System.nanoTime() > deadline) {
-                delegate.close();
-                throw new IOException("Connection timed out");
+        try {
```

Comment:
Would it make sense for this block to cover setOption for connectionTimeout above?

Reply:
yes